rclpy/test/test_client.py

```diff
@@ -15,9 +15,11 @@
 import platform
 import time
 import unittest
+import threading
```

Comment:
@KKSTB there are several flake8 errors, https://build.ros2.org/job/Rpr__rclpy__ubuntu_jammy_amd64/480/testReport/ could you address them?

Reply:
Is there any method to run flake8 checks locally? I got this exception on my computer:
```
 -- run_test.py: invoking following command in '/home/kk/ros2_rolling_ws/src/ros2/rclpy/rclpy':
53:  - /opt/ros/rolling/bin/ament_flake8 --xunit-file /home/kk/ros2_rolling_ws/build/rclpy/test_results/rclpy/flake8.xunit.xml
53: Traceback (most recent call last):
53:   File "/opt/ros/rolling/bin/ament_flake8", line 33, in <module>
53:     sys.exit(load_entry_point('ament-flake8==0.16.0', 'console_scripts', 'ament_flake8')())
53:   File "/home/kk/ros2_rolling_ws/build/ament_flake8/ament_flake8/main.py", line 31, in main
53:     rc, _ = main_with_errors(argv=argv)
53:   File "/home/kk/ros2_rolling_ws/build/ament_flake8/ament_flake8/main.py", line 88, in main_with_errors
53:     report = generate_flake8_report(
53:   File "/home/kk/ros2_rolling_ws/build/ament_flake8/ament_flake8/main.py", line 231, in generate_flake8_report
53:     opts, _ = parse_config_file(config_file)
53:   File "/home/kk/ros2_rolling_ws/build/ament_flake8/ament_flake8/main.py", line 199, in parse_config_file
53:     opts_manager = manager.OptionManager(
53: TypeError: OptionManager.__init__() missing 1 required keyword-only argument: 'formatter_names'
53: -- run_test.py: return code 1
53: -- run_test.py: generate result file '/home/kk/ros2_rolling_ws/build/rclpy/test_results/rclpy/flake8.xunit.xml' with failed test
53: -- run_test.py: verify result file '/home/kk/ros2_rolling_ws/build/rclpy/test_results/rclpy/flake8.xunit.xml'
53/57 Test #53: flake8 ............................***Failed    0.58 sec
```

Reply:
How about these?

```bash
cd <colcon_workspace>
colcon build --symlink-install --packages-select rclpy
colcon test --event-handlers console_direct+ --packages-select rclpy --ctest-args -R test_client
colcon test-result --verbose | less
```

Reply:
`colcon test --event-handlers console_direct+ --packages-select rclpy --ctest-args -R test_client` only do the test on `test_client`. Yet flake8 cannot be done locally.

Reply:
hmm, if i am not mistaken, i do that for my test to check flake8 warnings... anyway you addressed my comments, so i will start the CI.